src/Storages/ExternalStream/Kafka/KafkaSink.h

```diff
@@ -21,13 +21,14 @@ class ChunkSharder
     ChunkSharder(ExpressionActionsPtr sharding_expr_, const String & column_name);
     ChunkSharder();
 
-    BlocksWithShard shard(Block block, Int32 shard_cnt) const;
+    BlocksWithShard shard(Block block, Int32 shard_cnt);
 
 private:
-    Int32 getNextShardIndex(Int32 /*shard_cnt*/) const noexcept
+    Int32 getNextShardIndex(Int32 shard_cnt) noexcept
```

Comment:
Can you copy the comment from issue desc here to explain why we don't like librdkafka to decide the partition by using RD_KAFKA_PARTITION_UA